### PR TITLE
Update 'Upgrading Kubernetes' documentation

### PR DESCRIPTION
**Pull request by jossemargt-3pillar**
Problem: The Kubernetes upgrading process might be trivial but it is
easy to get lost in the overall process.

Solution: Update "Upgrading Kubernetes" documentation and add a process flow chart as a visual guide.

#### Proposed Changes ####

Adding a simplified visual help to RKE2 "Upgrading Kubernetes" documentation. We can see it as a sort of map.

#### Types of Changes ####

Documentation

#### Verification ####

No rke2 code has been touched.

#### Linked Issues ####

None

#### Further Comments ####

None

### CODE REVIEW

File: developer-docs/upgrading_kubernetes.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
 # Upgrade Kubernetes Process
 
-From time to time we need to update the version of Kubernetes used by RKE2. This document serves as a how-to for that process and constitutes a "release". The following steps are laid out in order.
+From time to time, we need to update the version of Kubernetes used by RKE2. This document serves as a how-to for said process and elaborates on what constitutes a "release". Although the steps described in the document are laid out in order, the overall flow looks this way:
 
-This process will need to be done any time a new release is needed.
+<!-- Source: https://gist.github.com/jossemargt-3pillar/4c93f23821a8cf9558692002d10c70ef -->
```
Comment (briandowns) [inline developer-docs/upgrading_kubernetes.md:5]: Is this line necessary?

Comment (jossemargt-3pillar) [reply developer-docs/upgrading_kubernetes.md:5]: No, it came from the initial iterations. Let me remove it.

File: developer-docs/upgrading_kubernetes.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
 # Upgrade Kubernetes Process
 
-From time to time we need to update the version of Kubernetes used by RKE2. This document serves as a how-to for that process and constitutes a "release". The following steps are laid out in order.
+From time to time, we need to update the version of Kubernetes used by RKE2. This document serves as a how-to for said process and elaborates on what constitutes a "release". Although the steps described in the document are laid out in order, the overall flow looks this way:
 
-This process will need to be done any time a new release is needed.
+![Upgrading kubernetes](upgrading_kubernetes.svg)
```
Comment (briandowns) [inline developer-docs/upgrading_kubernetes.md:5]:
```suggestion
![Upgrading Kubernetes](upgrading_kubernetes.svg)
```

File: developer-docs/upgrading_kubernetes.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...rc-41" transform="translate(1137.3828125, 106.75)"><rect style="" rx="25" ry="25" x="-77.6328125" y="-25" width="155.265625" height="50"/><g class="label" style="" transform="translate(-63.8828125, -17.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Cut RKE2</text></g><g xml:space="preserve" class="row"><text x="0" y="32">release candidate</text></g></g></g></g><g class="node default default" id="flowchart-cut_rke2_rc_rpm-43" transform="translate(1324.515625, 106.75)"><rect style="" rx="25" ry="25" x="-59.5" y="-25" width="119" height="50"/><g class="label" style="" transform="translate(-45.75, -17.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Cut RKE2</text></g><g xml:space="preserve" class="row"><text x="0" y="32">testing rpm's</text></g></g></g></g></g></g><g class="node default default" id="flowchart-prerelease-33" transform="translate(712.5078125, 25)"><rect style="" rx="17" ry="17" x="-101.46875" y="-17" width="202.9375" height="34"/><g class="label" style="" transform="translate(-89.71875, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Create cut release ticket</text></g></g></g></g></g></g></g></svg>
```
Comment (briandowns) [inline developer-docs/upgrading_kubernetes.svg:1]:
```suggestion
<svg id="mermaid-1643406172110" width="100%" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" height="987.25" style="max-width: 1413.52px; background-color: white;" viewBox="0 0 1413.515625 987.25"><style>#mermaid-1643406172110 {font-family:"trebuchet ms",verdana,arial,sans-serif;font-size:16px;fill:#333;}#mermaid-1643406172110 .error-icon{fill:#552222;}#mermaid-1643406172110 .error-text{fill:#552222;stroke:#552222;}#mermaid-1643406172110 .edge-thickness-normal{stroke-width:2px;}#mermaid-1643406172110 .edge-thickness-thick{stroke-width:3.5px;}#mermaid-1643406172110 .edge-pattern-solid{stroke-dasharray:0;}#mermaid-1643406172110 .edge-pattern-dashed{stroke-dasharray:3;}#mermaid-1643406172110 .edge-pattern-dotted{stroke-dasharray:2;}#mermaid-1643406172110 .marker{fill:#333333;stroke:#333333;}#mermaid-1643406172110 .marker.cross{stroke:#333333;}#mermaid-1643406172110 svg{font-family:"trebuchet ms",verdana,arial,sans-serif;font-size:16px;}#mermaid-1643406172110 .label{font-family:"trebuchet ms",verdana,arial,sans-serif;color:#333;}#mermaid-1643406172110 .cluster-label text{fill:#333;}#mermaid-1643406172110 .cluster-label span{color:#333;}#mermaid-1643406172110 .label text,#mermaid-1643406172110 span{fill:#333;color:#333;}#mermaid-1643406172110 .node rect,#mermaid-1643406172110 .node circle,#mermaid-1643406172110 .node ellipse,#mermaid-1643406172110 .node polygon,#mermaid-1643406172110 .node path{fill:#ECECFF;stroke:#9370DB;stroke-width:1px;}#mermaid-1643406172110 .node .label{text-align:center;}#mermaid-1643406172110 .node.clickable{cursor:pointer;}#mermaid-1643406172110 .arrowheadPath{fill:#333333;}#mermaid-1643406172110 .edgePath .path{stroke:#333333;stroke-width:2.0px;}#mermaid-1643406172110 .flowchart-link{stroke:#333333;fill:none;}#mermaid-1643406172110 .edgeLabel{background-color:#e8e8e8;text-align:center;}#mermaid-1643406172110 .edgeLabel rect{opacity:0.5;background-color:#e8e8e8;fill:#e8e8e8;}#mermaid-1643406172110 .cluster rect{fill:#ffffde;stroke:#aaaa33;stroke-width:1px;}#mermaid-1643406172110 .cluster text{fill:#333;}#mermaid-1643406172110 .cluster span{color:#333;}#mermaid-1643406172110 div.mermaidTooltip{position:absolute;text-align:center;max-width:200px;padding:2px;font-family:"trebuchet ms",verdana,arial,sans-serif;font-size:12px;background:hsl(80, 100%, 96.2745098039%);border:1px solid #aaaa33;border-radius:2px;pointer-events:none;z-index:100;}#mermaid-1643406172110 :root{--mermaid-font-family:"trebuchet ms",verdana,arial,sans-serif;}#mermaid-1643406172110 .default path{fill:#fff!important;stroke:#35b9ab!important;stroke-width:5px!important;}#mermaid-1643406172110 .default rect{fill:#fff!important;stroke:#35b9ab!important;stroke-width:5px!important;}#mermaid-1643406172110 .default polygon{fill:#fff!important;stroke:#35b9ab!important;stroke-width:5px!important;}#mermaid-1643406172110 .default ellipse{fill:#fff!important;stroke:#35b9ab!important;stroke-width:5px!important;}#mermaid-1643406172110 .default circle{fill:#fff!important;stroke:#35b9ab!important;stroke-width:5px!important;}#mermaid-1643406172110 .section path{fill:#fff!important;stroke:#bbb!important;stroke-width:1px!important;color:#173f4f!important;}#mermaid-1643406172110 .section rect{fill:#fff!important;stroke:#bbb!important;stroke-width:1px!important;color:#173f4f!important;}#mermaid-1643406172110 .section polygon{fill:#fff!important;stroke:#bbb!important;stroke-width:1px!important;color:#173f4f!important;}#mermaid-1643406172110 .section ellipse{fill:#fff!important;stroke:#bbb!important;stroke-width:1px!important;color:#173f4f!important;}#mermaid-1643406172110 .section circle{fill:#fff!important;stroke:#bbb!important;stroke-width:1px!important;color:#173f4f!important;}#mermaid-1643406172110 .section tspan{fill:#173f4f!important;}</style><g transform="translate(-6, 2)"><marker id="flowchart-pointEnd" class="marker flowchart" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="userSpaceOnUse" markerWidth="12" markerHeight="12" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowMarkerPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"/></marker><marker id="flowchart-pointStart" class="marker flowchart" viewBox="0 0 10 10" refX="0" refY="5" markerUnits="userSpaceOnUse" markerWidth="12" markerHeight="12" orient="auto"><path d="M 0 5 L 10 10 L 10 0 z" class="arrowMarkerPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"/></marker><marker id="flowchart-circleEnd" class="marker flowchart" viewBox="0 0 10 10" refX="11" refY="5" markerUnits="userSpaceOnUse" markerWidth="11" markerHeight="11" orient="auto"><circle cx="5" cy="5" r="5" class="arrowMarkerPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"/></marker><marker id="flowchart-circleStart" class="marker flowchart" viewBox="0 0 10 10" refX="-1" refY="5" markerUnits="userSpaceOnUse" markerWidth="11" markerHeight="11" orient="auto"><circle cx="5" cy="5" r="5" class="arrowMarkerPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"/></marker><marker id="flowchart-crossEnd" class="marker cross flowchart" viewBox="0 0 11 11" refX="12" refY="5.2" markerUnits="userSpaceOnUse" markerWidth="11" markerHeight="11" orient="auto"><path d="M 1,1 l 9,9 M 10,1 l -9,9" class="arrowMarkerPath" style="stroke-width: 2; stroke-dasharray: 1, 0;"/></marker><marker id="flowchart-crossStart" class="marker cross flowchart" viewBox="0 0 11 11" refX="-1" refY="5.2" markerUnits="userSpaceOnUse" markerWidth="11" markerHeight="11" orient="auto"><path d="M 1,1 l 9,9 M 10,1 l -9,9" class="arrowMarkerPath" style="stroke-width: 2; stroke-dasharray: 1, 0;"/></marker><g class="root"><g class="clusters"/><g class="edgePaths"><path d="M712.5078125,42L712.5078125,49.083333333333336C712.5078125,56.166666666666664,712.5078125,70.33333333333333,712.5078125,84.5C712.5078125,98.66666666666667,712.5078125,112.83333333333333,712.5078125,119.91666666666667L712.5078125,127" id="L-prerelease-rc-0" class=" edge-thickness-normal edge-pattern-dotted flowchart-link LS-prerelease LE-rc" style="fill:none;stroke-width:2px;stroke-dasharray:3;" marker-end="url(#flowchart-pointEnd)"/><path d="M712.5078125,321.25L712.5078125,325.4166666666667C712.5078125,329.5833333333333,712.5078125,337.9166666666667,712.5078125,346.25C712.5078125,354.5833333333333,712.5078125,362.9166666666667,712.5078125,367.0833333333333L712.5078125,371.25" id="L-rc-kdm_rc-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-rc LE-kdm_rc" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/><path d="M712.5078125,475.25L712.5078125,481C712.5078125,486.75,712.5078125,498.25,712.5078125,509.75C712.5078125,521.25,712.5078125,532.75,712.5078125,538.5L712.5078125,544.25" id="L-kdm_rc-primary_release-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-kdm_rc LE-primary_release" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/><path d="M712.5078125,656.25L712.5078125,660.4166666666666C712.5078125,664.5833333333334,712.5078125,672.9166666666666,712.5078125,681.25C712.5078125,689.5833333333334,712.5078125,697.9166666666666,712.5078125,702.0833333333334L712.5078125,706.25" id="L-primary_release-kdm_stable-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-primary_release LE-kdm_stable" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/><path d="M712.5078125,810.25L712.5078125,816C712.5078125,821.75,712.5078125,833.25,712.5078125,844.75C712.5078125,856.25,712.5078125,867.75,712.5078125,873.5L712.5078125,879.25" id="L-kdm_stable-stable_release-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-kdm_stable LE-stable_release" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/></g><g class="edgeLabels"><g class="edgeLabel" transform="translate(712.5078125, 84.5)"><g class="label" transform="translate(-89.0703125, -17.5)"><rect rx="0" ry="0" width="178.06985473632812" height="34.029537200927734"/><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Once Kubernetes release</text></g><g xml:space="preserve" class="row"><text x="0" y="32">is generally available</text></g></g></g></g><g class="edgeLabel"><g class="label" transform="translate(0, 0)"><rect rx="0" ry="0" width="0" height="0"/><g style=""><g xml:space="preserve" class="row"/></g></g></g><g class="edgeLabel" transform="translate(712.5078125, 509.75)"><g class="label" transform="translate(-154.3125, -9.5)"><rect rx="0" ry="0" width="308.47412109375" height="18.029537200927734"/><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Once QA approves RKE2 release candidates</text></g></g></g></g><g class="edgeLabel"><g class="label" transform="translate(0, 0)"><rect rx="0" ry="0" width="0" height="0"/><g style=""><g xml:space="preserve" class="row"/></g></g></g><g class="edgeLabel" transform="translate(712.5078125, 844.75)"><g class="label" transform="translate(-116.4296875, -9.5)"><rect rx="0" ry="0" width="232.75006103515625" height="18.029537200927734"/><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Once QA approves RKE2 releases</text></g></g></g></g></g><g class="nodes"><g class="root" transform="translate(403.9609375, 871.25)"><g class="clusters"><g class="cluster section section" id="stable_release"><rect style="" rx="0" ry="0" x="8" y="8" width="602.09375" height="104"/><g class="cluster-label" transform="translate(217.640625, 13)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">RKE2 Stable Release Flow</text></g></g></g></g></g><g class="edgePaths"><path d="M203.109375,60L207.27604166666666,60C211.44270833333334,60,219.77604166666666,60,228.109375,60C236.44270833333334,60,244.77604166666666,60,248.94270833333334,60L253.109375,60" id="L-cut_rke_stable_rpm-rke_stable-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-cut_rke_stable_rpm LE-rke_stable" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/></g><g class="edgeLabels"><g class="edgeLabel"><g class="label" transform="translate(0, 0)"><rect rx="0" ry="0" width="0" height="0"/><g style=""><g xml:space="preserve" class="row"/></g></g></g></g><g class="nodes"><g class="node default default" id="flowchart-rke_stable-55" transform="translate(419.1015625, 60)"><rect style="" rx="17" ry="17" x="-165.9921875" y="-17" width="331.984375" height="34"/><g class="label" style="" transform="translate(-154.2421875, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Update RKE2 channel server configurations</text></g></g></g></g><g class="node default default" id="flowchart-cut_rke_stable_rpm-54" transform="translate(118.0546875, 60)"><rect style="" rx="17" ry="17" x="-85.0546875" y="-17" width="170.109375" height="34"/><g class="label" style="" transform="translate(-73.3046875, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Cut RKE2 stable rpm</text></g></g></g></g></g></g><g class="root" transform="translate(419.9765625, 698.25)"><g class="clusters"><g class="cluster section section" id="kdm_stable"><rect style="" rx="0" ry="0" x="8" y="8" width="570.0625" height="104"/><g class="cluster-label" transform="translate(172.734375, 13)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Rancher KDM Stable Release Flow</text></g></g></g></g></g><g class="edgePaths"><path d="M278.140625,60L282.3072916666667,60C286.4739583333333,60,294.8072916666667,60,303.140625,60C311.4739583333333,60,319.8072916666667,60,323.9739583333333,60L328.140625,60" id="L-kdm_channel_stable-kdm_generated_data_stable-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-kdm_channel_stable LE-kdm_generated_data_stable" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/></g><g class="edgeLabels"><g class="edgeLabel"><g class="label" transform="translate(0, 0)"><rect rx="0" ry="0" width="0" height="0"/><g style=""><g xml:space="preserve" class="row"/></g></g></g></g><g class="nodes"><g class="node default default" id="flowchart-kdm_generated_data_stable-53" transform="translate(440.6015625, 60)"><rect style="" rx="17" ry="17" x="-112.4609375" y="-17" width="224.921875" height="34"/><g class="label" style="" transform="translate(-100.7109375, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Update KDM generated data</text></g></g></g></g><g class="node default default" id="flowchart-kdm_channel_stable-52" transform="translate(155.5703125, 60)"><rect style="" rx="17" ry="17" x="-122.5703125" y="-17" width="245.140625" height="34"/><g class="label" style="" transform="translate(-110.8203125, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Update RKE release references</text></g></g></g></g></g></g><g class="root" transform="translate(117.875, 536.25)"><g class="clusters"><g class="cluster section section" id="primary_release"><rect style="" rx="0" ry="0" x="8" y="8" width="1166.265625" height="104"/><g class="cluster-label" transform="translate(494.7578125, 13)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">RKE2 Primary Release Flow</text></g></g></g></g></g><g class="edgePaths"><path d="M238.59375,60L242.76041666666666,60C246.92708333333334,60,255.26041666666666,60,263.59375,60C271.9270833333333,60,280.2604166666667,60,284.4270833333333,60L288.59375,60" id="L-cut_rke-cut_rke2_latest_rpm-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-cut_rke LE-cut_rke2_latest_rpm" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/><path d="M465.15625,60L469.3229166666667,60C473.4895833333333,60,481.8229166666667,60,490.15625,60C498.4895833333333,60,506.8229166666667,60,510.9895833333333,60L515.15625,60" id="L-cut_rke2_latest_rpm-update_release_notes-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-cut_rke2_latest_rpm LE-update_release_notes" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/><path d="M690.859375,60L703.21875,60C715.578125,60,740.296875,60,765.015625,60C789.734375,60,814.453125,60,826.8125,60L839.171875,60" id="L-update_release_notes-GA-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-update_release_notes LE-GA" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/></g><g class="edgeLabels"><g class="edgeLabel"><g class="label" transform="translate(0, 0)"><rect rx="0" ry="0" width="0" height="0"/><g style=""><g xml:space="preserve" class="row"/></g></g></g><g class="edgeLabel"><g class="label" transform="translate(0, 0)"><rect rx="0" ry="0" width="0" height="0"/><g style=""><g xml:space="preserve" class="row"/></g></g></g><g class="edgeLabel" transform="translate(765.015625, 60)"><g class="label" transform="translate(-49.15625, -17.5)"><rect rx="0" ry="0" width="98.28914642333984" height="34.029537200927734"/><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">At least 24hrs</text></g><g xml:space="preserve" class="row"><text x="0" y="32">have passed</text></g></g></g></g></g><g class="nodes"><g class="node default default" id="flowchart-cut_rke2_latest_rpm-47" transform="translate(376.875, 60)"><rect style="" rx="17" ry="17" x="-88.28125" y="-17" width="176.5625" height="34"/><g class="label" style="" transform="translate(-76.53125, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Cut RKE2 latest rpm's</text></g></g></g></g><g class="node default default" id="flowchart-cut_rke-46" transform="translate(135.796875, 60)"><rect style="" rx="17" ry="17" x="-102.796875" y="-17" width="205.59375" height="34"/><g class="label" style="" transform="translate(-91.046875, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Cut RKE2 primary release</text></g></g></g></g><g class="node default default" id="flowchart-update_release_notes-49" transform="translate(603.0078125, 60)"><rect style="" rx="17" ry="17" x="-87.8515625" y="-17" width="175.703125" height="34"/><g class="label" style="" transform="translate(-76.1015625, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Update release notes</text></g></g></g></g><g class="node default default" id="flowchart-GA-51" transform="translate(994.21875, 60)"><rect style="" rx="17" ry="17" x="-155.046875" y="-17" width="310.09375" height="34"/><g class="label" style="" transform="translate(-143.296875, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Set RKE2 releases as generally available</text></g></g></g></g></g></g><g class="root" transform="translate(408.125, 363.25)"><g class="clusters"><g class="cluster section section" id="kdm_rc"><rect style="" rx="0" ry="0" x="8" y="8" width="593.765625" height="104"/><g class="cluster-label" transform="translate(197.546875, 13)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Rancher KDM RC Release Flow</text></g></g></g></g></g><g class="edgePaths"><path d="M301.84375,60L306.0104166666667,60C310.1770833333333,60,318.5104166666667,60,326.84375,60C335.1770833333333,60,343.5104166666667,60,347.6770833333333,60L351.84375,60" id="L-kdm_channel_rc-kdm_generated_data_rc-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-kdm_channel_rc LE-kdm_generated_data_rc" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/></g><g class="edgeLabels"><g class="edgeLabel"><g class="label" transform="translate(0, 0)"><rect rx="0" ry="0" width="0" height="0"/><g style=""><g xml:space="preserve" class="row"/></g></g></g></g><g class="nodes"><g class="node default default" id="flowchart-kdm_generated_data_rc-45" transform="translate(464.3046875, 60)"><rect style="" rx="17" ry="17" x="-112.4609375" y="-17" width="224.921875" height="34"/><g class="label" style="" transform="translate(-100.7109375, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Update KDM generated data</text></g></g></g></g><g class="node default default" id="flowchart-kdm_channel_rc-44" transform="translate(167.421875, 60)"><rect style="" rx="17" ry="17" x="-134.421875" y="-17" width="268.84375" height="34"/><g class="label" style="" transform="translate(-122.671875, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Update RKE RC release references</text></g></g></g></g></g></g><g class="root" transform="translate(0.5, 119)"><g class="clusters"><g class="cluster section section" id="rc"><rect style="" rx="0" ry="0" x="8" y="8" width="1401.015625" height="186.25"/><g class="cluster-label" transform="translate(607.671875, 13)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">RKE2 Release Candidate Flow</text></g></g></g></g></g><g class="edgePaths"><path d="M197.65625,86.99730324577952L210.85286458333334,83.83108603814959C224.04947916666666,80.66486883051968,250.44270833333334,74.33243441525984,276.8359375,71.16621720762991C303.2291666666667,68,329.6223958333333,68,342.8190104166667,68L356.015625,68" id="L-cut_k8s-update_kubeproxy-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-cut_k8s LE-update_kubeproxy" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/><path d="M514.09375,68L528.7994791666666,68C543.5052083333334,68,572.9166666666666,68,602.328125,70.94423252872635C631.7395833333334,73.8884650574527,661.1510416666666,79.77693011490541,675.8567708333334,82.72116264363176L690.5625,85.66539517235812" id="L-update_kubeproxy-update_rke_refs-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-update_kubeproxy LE-update_rke_refs" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/><path d="M197.65625,126.50269675422048L210.85286458333334,129.6689139618504C224.04947916666666,132.83513116948032,250.44270833333334,139.16756558474017,290.0091145833333,142.33378279237007C329.5755208333333,145.5,382.3151041666667,145.5,436.5638020833333,145.5C490.8125,145.5,546.5703125,145.5,589.1549479166666,142.55576747127364C631.7395833333334,139.6115349425473,661.1510416666666,133.7230698850946,675.8567708333334,130.77883735636826L690.5625,127.83460482764188" id="L-cut_k8s-update_rke_refs-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-cut_k8s LE-update_rke_refs" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/><path d="M901.1875,106.75L914.4010416666666,106.75C927.6145833333334,106.75,954.0416666666666,106.75,980.46875,106.75C1006.8958333333334,106.75,1033.3229166666667,106.75,1046.5364583333333,106.75L1059.75,106.75" id="L-update_rke_refs-cut_rke2_rc-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-update_rke_refs LE-cut_rke2_rc" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/><path d="M1215.015625,106.75L1219.1822916666667,106.75C1223.3489583333333,106.75,1231.6822916666667,106.75,1240.015625,106.75C1248.3489583333333,106.75,1256.6822916666667,106.75,1260.8489583333333,106.75L1265.015625,106.75" id="L-cut_rke2_rc-cut_rke2_rc_rpm-0" class=" edge-thickness-normal edge-pattern-solid flowchart-link LS-cut_rke2_rc LE-cut_rke2_rc_rpm" style="fill:none;" marker-end="url(#flowchart-pointEnd)"/></g><g class="edgeLabels"><g class="edgeLabel" transform="translate(276.8359375, 68)"><g class="label" transform="translate(-54.1796875, -17.5)"><rect rx="0" ry="0" width="108.31807708740234" height="34.029537200927734"/><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">For Kubernetes</text></g><g xml:space="preserve" class="row"><text x="0" y="32">1.20 and 1.21</text></g></g></g></g><g class="edgeLabel" transform="translate(602.328125, 68)"><g class="label" transform="translate(-63.234375, -17.5)"><rect rx="0" ry="0" width="126.40396118164062" height="34.029537200927734"/><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Once RKE2-charts</text></g><g xml:space="preserve" class="row"><text x="0" y="32">index refreshes</text></g></g></g></g><g class="edgeLabel" transform="translate(435.0546875, 145.5)"><g class="label" transform="translate(-60.8203125, -17.5)"><rect rx="0" ry="0" width="121.5866928100586" height="34.029537200927734"/><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">From Kubernetes</text></g><g xml:space="preserve" class="row"><text x="0" y="32">1.22 onwards</text></g></g></g></g><g class="edgeLabel" transform="translate(980.46875, 106.75)"><g class="label" transform="translate(-54.28125, -17.5)"><rect rx="0" ry="0" width="108.51527404785156" height="34.029537200927734"/><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Once RKE2 PR's</text></g><g xml:space="preserve" class="row"><text x="0" y="32">are approved</text></g></g></g></g><g class="edgeLabel"><g class="label" transform="translate(0, 0)"><rect rx="0" ry="0" width="0" height="0"/><g style=""><g xml:space="preserve" class="row"/></g></g></g></g><g class="nodes"><g class="node default default" id="flowchart-update_kubeproxy-35" transform="translate(435.0546875, 68)"><rect style="" rx="25" ry="25" x="-79.0390625" y="-25" width="158.078125" height="50"/><g class="label" style="" transform="translate(-65.2890625, -17.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Update kubeproxy</text></g><g xml:space="preserve" class="row"><text x="0" y="32">in RKE2-charts</text></g></g></g></g><g class="node default default" id="flowchart-cut_k8s-34" transform="translate(115.328125, 106.75)"><rect style="" rx="25" ry="25" x="-82.328125" y="-25" width="164.65625" height="50"/><g class="label" style="" transform="translate(-68.578125, -17.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Cut hardened</text></g><g xml:space="preserve" class="row"><text x="0" y="32">Kubernetes release</text></g></g></g></g><g class="node default default" id="flowchart-update_rke_refs-37" transform="translate(795.875, 106.75)"><rect style="" rx="25" ry="25" x="-105.3125" y="-25" width="210.625" height="50"/><g class="label" style="" transform="translate(-91.5625, -17.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Update RKE2 components</text></g><g xml:space="preserve" class="row"><text x="0" y="32">version references</text></g></g></g></g><g class="node default default" id="flowchart-cut_rke2_rc-41" transform="translate(1137.3828125, 106.75)"><rect style="" rx="25" ry="25" x="-77.6328125" y="-25" width="155.265625" height="50"/><g class="label" style="" transform="translate(-63.8828125, -17.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Cut RKE2</text></g><g xml:space="preserve" class="row"><text x="0" y="32">release candidate</text></g></g></g></g><g class="node default default" id="flowchart-cut_rke2_rc_rpm-43" transform="translate(1324.515625, 106.75)"><rect style="" rx="25" ry="25" x="-59.5" y="-25" width="119" height="50"/><g class="label" style="" transform="translate(-45.75, -17.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Cut RKE2</text></g><g xml:space="preserve" class="row"><text x="0" y="32">testing rpm's</text></g></g></g></g></g></g><g class="node default default" id="flowchart-prerelease-33" transform="translate(712.5078125, 25)"><rect style="" rx="17" ry="17" x="-101.46875" y="-17" width="202.9375" height="34"/><g class="label" style="" transform="translate(-89.71875, -9.5)"><g style=""><g xml:space="preserve" class="row"><text x="0" y="16">Create cut release ticket</text></g></g></g></g></g></g></g></svg>
```

File: developer-docs/upgrading_kubernetes.md [hunk 4]
```diff
@@ -74,7 +74,7 @@ Once QA signs off on the RC, it's time to cut the primary release. Go to the [rk
 * Enter the desired version into the "Tag version" box. 
     * Example tag: `v1.21.4+rke2r1`
 
-Leave the release as "prerelease". This will be unchecked as soon as CI completes successfully.
+Ensure "prerelease" checkbox is selected. This will be unchecked as soon as CI completes successfully.
```
Comment (briandowns) [inline developer-docs/upgrading_kubernetes.md:77]:
```suggestion
Ensure "prerelease" checkbox is selected.
```

File: developer-docs/upgrading_kubernetes.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
 # Upgrade Kubernetes Process
 
-From time to time we need to update the version of Kubernetes used by RKE2. This document serves as a how-to for that process and constitutes a "release". The following steps are laid out in order.
+From time to time, we need to update the version of Kubernetes used by RKE2. This document serves as a how-to for said process and elaborates on what constitutes a "release". Although the steps described in the document are laid out in order, the overall flow looks this way:
 
-This process will need to be done any time a new release is needed.
+![Upgrading Kubernetes](upgrading_kubernetes.svg)
 
-A handy checklist can be found [here](#release-checklist).
+**NOTE:** This process will be needed whenever a new release is required, even when no new kubernetes release is available. A handy checklist can be found [here](#release-process).
```
Comment (briandowns) [inline developer-docs/upgrading_kubernetes.md:7]:
```suggestion
**NOTE:** This process will be needed whenever a new release is required, even when no new Kubernetes release is available. A handy checklist can be found [here](#release-process).
```

File: developer-docs/upgrading_kubernetes.md [hunk 6]
```diff
@@ -1,10 +1,10 @@
 # Upgrade Kubernetes Process
 
-From time to time we need to update the version of Kubernetes used by RKE2. This document serves as a how-to for that process and constitutes a "release". The following steps are laid out in order.
+From time to time, we need to update the version of Kubernetes used by RKE2. This document serves as a how-to for said process and elaborates on what constitutes a "release". Although the steps described in the document are laid out in order, the overall flow looks this way:
 
-This process will need to be done any time a new release is needed.
+![Upgrading Kubernetes](upgrading_kubernetes.svg)
```
Comment (briandowns) [inline developer-docs/upgrading_kubernetes.md:5]: I don't know if the top of the document is the best place for the image.

Comment (jossemargt-3pillar) [reply developer-docs/upgrading_kubernetes.md:5]: I did it imitating [Kubernetes contributors documentation](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md) layout, also thinking on newcomers that rather want to see an image before jumping on the docs. Either way, I can move it where you see it fit

Comment (briandowns) [reply developer-docs/upgrading_kubernetes.md:5]: Let's move it down to the bottom but right above the "Release Process" section. I think where at the point now that a table of contents would be extremely helpful.

File: developer-docs/upgrading_kubernetes.svg (hunk 3 above)
Comment (jossemargt-3pillar) [reply developer-docs/upgrading_kubernetes.svg:1]: Can I ask what did you change here? I checked these changes locally and see the resulting svg like the old one

File: developer-docs/upgrading_kubernetes.md (hunk 6 above)
Comment (jossemargt-3pillar) [reply developer-docs/upgrading_kubernetes.md]: I generated a TOC, please take a look

**Review by dereknola — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Pull request merged**